nncf/quantization/layers.py

```diff
@@ -230,17 +240,40 @@ def __init__(self, config):
         self.collect_scale_statistics = False
         if self.per_channel:
             self.scale_shape = get_per_channel_scale_shape(self.input_shape, self.is_weights)
-        self.scale = nn.Parameter(torch.ones(self.scale_shape), requires_grad=True)
-        self.eps = 1e-16
+        self.scale_tensor = nn.Parameter(torch.ones(self.scale_shape), requires_grad=True)
+        if self.scale_log_flag:
+            self.scale_tensor.data.log_()
+            self.eps = 0
+        else:
+            self.eps = 1e-16
         if config.signedness_to_force is not None:
             self.signed = int(config.signedness_to_force)
         self.set_level_ranges()
 
+        #register hooks to convert scale from log vertion to original vertion during load and save
+        def hook_fn_load_state_dict(state_dict, prefix, local_metadata, strict, missing_keys, unexpected_keys, error_msgs):
```

Comment:
How about redefining the `torch.nn.Module._load_state_dict` and `torch.nn.Module._state_dict` functions instead of using the hook mechanism?

Reply:
Orignal BaseQuantizer (see top of the file) use the hook. I just used the same approach for consistency.

Reply:
Don't think it's a good idea to redefine standard torch functions. The behavior can be changed in the newer version, and our redefined implementation would lead to errors and bugs. API with hooks is the way, suggested and supported by PyTorch, to extend `load_state_dict` functionality. IMHO, it is better to stick to this approach